### PR TITLE
ffi: support mesh registration

### PR DESCRIPTION
**Pull request by kvark**
cc @lpenguin 
Note the `Transform` change here as well as the new API entry points.

### CODE REVIEW

**Comment by kvark** [comment]
@lpenguin I think this covers all of https://github.com/KranX/Vangers/issues/533 now, with a few corrections. Namely, method names and the extra `color_id` argument.

**Comment by lpenguin** [comment]
@kvark Nice, thanks!

File: lib/ffi/src/lib.rs [hunk 1]
```diff
@@ -401,3 +475,120 @@ pub extern "C" fn rv_render(ctx: &mut Context, viewport: Rect) {
 
     ctx.queue.submit(Some(encoder.finish()));
 }
+
+fn vec_i2f(v: [i32; 3]) -> [f32; 3] {
+    [v[0] as f32, v[1] as f32, v[2] as f32]
+}
+
+#[no_mangle]
+pub extern "C" fn rn_model_create(
```
Comment (lpenguin) [inline lib/ffi/src/lib.rs:484]: Should it be `rv_` names instead `rn_`?

Comment (kvark) [reply lib/ffi/src/lib.rs]: funny how errors proliferate and multiply themselves!

**Pull request merged**